src/iOS/Services/iOSPushNotificationService.cs

```diff
@@ -19,6 +19,15 @@ public Task<string> GetTokenAsync()
 
         public bool IsRegisteredForPush => UIApplication.SharedApplication.IsRegisteredForRemoteNotifications;
 
+        public async Task<bool> GetNotificationsSettingsEnabledAsync()
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            UNUserNotificationCenter.Current.GetNotificationSettings((settings) => {
+                tcs.SetResult(settings.AlertSetting == UNNotificationSetting.Enabled);
+            });
+            return await tcs.Task;
+        }
```

Comment:
Use the async version of the method: `UNUserNotificationCenter.Current.GetNotificationSettingsAsync()` so that you don't have to use the `TaskCompletionSource<bool>`.
And just FYI, in here you don't gain anything by async awaiting so the `async` is not needed and you should return directly `tcs.Task`.
Nonetheless, it's better to use directly the async version of the method:
``` c#
public async Task<bool> GetNotificationsSettingsEnabledAsync()
{
    var settings = await UNUserNotificationCenter.Current.GetNotificationSettingsAsync();
    return settings.AlertSetting == UNNotificationSetting.Enabled;
}
```